### ISSUE TITLE
Add client-side error monitoring

### ISSUE THREAD

**Issue by nickytonline** [forem/forem#2501]
**Is your feature request related to a problem? Please describe.**

This is not a user facing issue directly, but errors that are not caught on the front-end can damage a user's experience if no one is aware of it.

**Describe the solution you'd like**

Consider a client-side monitoring tool. There are many, but I just happened to notice that Sentry has a [plan for great open source projects](https://sentry.io/for/open-source/). I'm not partial to any solution, but I have used Sentry in the past on projects and had good experiences with it.

**Describe alternatives you've considered**

We currently console log in some areas of the application, and that's about it, e.g. https://github.com/thepracticaldev/dev.to/blob/master/app/javascript/src/utils/sendFollowUser.js#L23

![Windows Blue Screen of Death](https://media.giphy.com/media/KlrMS4vyq5KSY/giphy.gif)

**Comment by rhymes** [comment]
Server side they use Airbrake to track errors, which provides [airbrake-js](https://github.com/airbrake/airbrake-js) that also installs a global handler in case of uncaught errors :-) 

It's similar to Sentry's toolkit I guess

**Comment by nickytonline** [comment]
Airbrake probably makes sense then on the client-side as well to keep things uniform.

**Comment by maestromac** [comment]
Yep. We should enable Airbrake in the preact side.

**Comment by nickytonline** [comment]
I'll take this on and may synch up with you @maestromac.

**Comment by nickytonline** [comment]
Looks like the airbrake JS client is already installed. See https://github.com/thepracticaldev/dev.to/blob/master/app/assets/javascripts/base.js.erb#L683. However, it's only loaded when it's running production only. It would make sense to make it available in all environments, e.g. dev, so that devs contributing can add error reporting via airbrake in PRs. From what I can see on airbrake.io, there is no OSS license, so not sure how contributors could get a license for dev.

**Comment by rhymes** [comment]
@nickytonline I think it's enabled only in production because there's no OSS/free tier unlike Sentry. 😞

Maybe a section of the docs explaining how to add it to the code would be already something. 

A few notes about Airbrake JS current state:

* it's sampled on the client side, so not all JS errors are sent, which can be an issue, because being random at a global level it means that a meaningful error is never logged:

https://github.com/thepracticaldev/dev.to/blob/398aa71ec30df406960d73c5114dce8b687a97b9/app/assets/javascripts/base.js.erb#L690-L696

* it's directly used only once but re-initialized again (not sure why):

https://github.com/thepracticaldev/dev.to/blob/398aa71ec30df406960d73c5114dce8b687a97b9/app/assets/javascripts/utilities/getCsrfToken.js.erb#L13-L18

Basically now it's catching uncaught errors and notifying them with a sampler. 

If I'm not mistaken by your article explaining the frontend, Airbrake is already initialized on the Preact side, at least for uncaught JS errors, since it's enabled on this side.

You can read more about its usage on the [airbrake-js repo](https://github.com/airbrake/airbrake-js#basic-usage) and this is the doc to use it with React's [components](https://airbrake.io/docs/installing-airbrake/installing-airbrake-in-a-react-app/)

**Comment by maestromac** [comment]
@rhymes the re-initialize portion could be a mistake on our part.

I'll bring this up to Ben and see what he thinks about Sentry.

**Comment by nickytonline** [comment]
Not stale

**Comment by cmgorton** [comment]
@nickytonline is this something we are likely to work on? There has been further discussion on this in over a year so wondering if I can close this issue?

**Comment by nickytonline** [comment]
@cmgorton, closing this as we have [Honeybadger](https://www.honeybadger.io/) in place for this.
